libraries/AP_Common/Location.cpp

```diff
@@ -355,9 +355,11 @@ bool Location::sanitize(const Location &defaultLoc)
 
     // convert relative alt=0 to mean current alt
     if (alt == 0 && relative_alt) {
-        relative_alt = false;
-        alt = defaultLoc.alt;
-        has_changed = true;
+        int32_t defaultLoc_alt;
+        if (defaultLoc.get_alt_cm(get_alt_frame(), defaultLoc_alt)) {
+            alt = defaultLoc_alt;
```

Comment:
```suggestion
        if (defaultLoc.get_alt_cm(get_alt_frame(), alt)) {
```
I think this would work?